src/Faker/Provider/es_MX/Person.php

```diff
@@ -0,0 +1,337 @@
+    }
+    protected static function removePrefixes($name)
+    {
+        foreach (self::$prefixes as $prefix) {
+            $name = preg_replace($prefix, '', $name);
+        }
+        return $name;
+    }
+    protected static function removeAccents($name)
+    {
+        return str_replace(array_keys(self::$accents), array_values(self::$accents), $name);
+    }
+
+    protected static $notAcceptedNames = array(
+        '/^MARIA DEL /',
+        '/^MARIA DE LOS /',
+        '/^MARIA /',
+        '/^JOSE DE /',
+        '/^JOSE /',
+        '/^MA. /',
+        '/^MA /',
+        '/^M. /',
+        '/^J. /',
+        '/^J /'
+    );
+
+    protected static $prefixes = array(
+        '/^DE /',
+        '/^DEL /'
+    );
+
+    protected static $badWordsCURP = array(
+        "BACA" => "BXCA",
+        "LOCO" => "LXCO",
+        "BAKA" => "BXKA",
+        "BUEI" => "BXEI",
+        "BUEY" => "BXEY",
+        "CACA" => "CXCA",
+        "CACO" => "CXCO",
+        "CAGA" => "CXGA",
+        "CAGO" => "CXGO",
+        "CAKA" => "CXKA",
+        "CAKO" => "CXKO",
+        "COGE" => "CXGE",
+        "COGI" => "CXGI",
+        "COJA" => "CXJA",
+        "COJE" => "CXJE",
+        "COJI" => "CXJI",
+        "COJO" => "CXJO",
+        "COLA" => "CXLA",
+        "CULO" => "CXLO",
+        "FALO" => "FXLO",
+        "FETO" => "FXTO",
+        "GETA" => "GXTA",
+        "GUEI" => "GXEI",
+        "GUEY" => "GXEY",
+        "JETA" => "JXTA",
+        "JOTO" => "JXTO",
+        "KACA" => "KXCA",
+        "KACO" => "KXCO",
+        "KAGA" => "KXGA",
+        "KAGO" => "KXGO",
+        "KAKA" => "KXKA",
+        "KAKO" => "KXKO",
+        "KOGE" => "KXGE",
+        "KOGI" => "KXGI",
+        "KOJA" => "KXJA",
+        "KOJE" => "KXJE",
+        "KOJI" => "KXJI",
+        "KOJO" => "KXJO",
+        "KOLA" => "KXLA",
+        "KULO" => "KXLO",
+        "LILO" => "LXLO",
+        "LOKA" => "LXKA",
+        "LOKO" => "LXKO",
+        "MAME" => "MXME",
+        "MAMO" => "MXMO",
+        "MEAR" => "MXAR",
+        "MEAS" => "MXAS",
+        "MEON" => "MXON",
+        "MIAR" => "MXAR",
+        "MION" => "MXON",
+        "MOCO" => "MXCO",
+        "MOKO" => "MXKO",
+        "MULA" => "MXLA",
+        "MULO" => "MXLO",
+        "NACA" => "NXCA",
+        "NACO" => "NXCO",
+        "PEDA" => "PXDA",
+        "PEDO" => "PXDO",
+        "PENE" => "PXNE",
+        "PIPI" => "PXPI",
+        "PITO" => "PXTO",
+        "POPO" => "PXPO",
+        "PUTA" => "PXTA",
+        "PUTO" => "PXTO",
+        "QULO" => "QXLO",
+        "RATA" => "RXTA",
+        "ROBA" => "RXBA",
+        "ROBE" => "RXBE",
+        "ROBO" => "RXBO",
+        "RUIN" => "RXIN",
+        "SENO" => "SXNO",
+        "TETA" => "TXTA",
+        "VACA" => "VXCA",
+        "VAGA" => "VXGA",
+        "VAGO" => "VXGO",
+        "VAKA" => "VXKA",
+        "VUEI" => "VXEI",
+        "VUEY" => "VXEY",
+        "WUEI" => "WXEI",
+        "WUEY" => "WXEY"
+    );
+
+    protected static $badWordsRFC = array(
```

Comment:
Those words are offensive, so they are on the list.

Example:
**Ca**rolina **C**amacho **A**lvarez
**RFC:** CACA (SHIT).

That words, don't appear in valid RFC code.

Reference:

 - Unam. (2016). ¿Cómo formar el RFC?. Retrieved 11 October, 2016, from http://cec.cele.unam.mx/include/howToRFC.php
 - (2014). El RFC hace referencia a una palabra altisonante o
inconveniente. Manual Armado RFC, 4. Retrieved from:http://www.betterhouse.com.mx/inicio/images/capacitacion/bnt04.pdf